### DISCUSSION TITLE
URGENT: Please help

### DISCUSSION THREAD

**Discussion by b9Joker108** [CodelyTV/dotly#268] [Q&A] [answered]
I just installed dotly, but I didn't know it was going to overwrite my .zshrc config file. I put so much work into that config file, over about two years. I am really upset that I have potentially lost all that work. Did the installation of dotly somehow create a backup of it or did it just overwrite all that work? Also, installing Dotly has broken my Termux, apt and pkg don't work.

If installing dotly does that you really should have a very clear warning at the outset of the README.md!

I made this Reddit post about this:

https://www.reddit.com/r/termux/s/4ceuX8xCZj

**Comment by rgomezcasas** [comment, marked as answer]
Hey,

Yes, there is a backup in your `$HOME/.pre_dotly_backup`. [The code that does the backup is this](https://github.com/CodelyTV/dotly/blob/main/scripts/self/utils/install.sh#L41) if you want to add some output so other people knows about the backup.

Regards.

**Comment by b9Joker108** [comment]
This installation of Dotly has caused me such stress, I was shaking and in tears. I luckily found a version of my .zshrc configuration file in my guest PRoot-distro alias of Arch Linux, from a month ago, and overwrote the Dotly .zshrc file with it, and my broken Termux started working again, and I had my environment and workflow back, well mostly. 

Dotly, apart from totally replacing the Zsh config, the Dotly .zshrc totally redefines the environment Path variable, which breaks Termux. Isn't there some way that the installation of Dotly can be tweaked so it integrates with the current user's host configuration files? That is, it appends code and annotations to the user's host configuration files, rather than replacing and archiving them? I feel that for most use-cases, that would be more functional, beneficial and practical, as most users who install Dotly, won't be doing so on a new or fresh system install, and they will most probably, already have developed Zsh and Bash configuration files, like I have.

You cannot imagine how relieved I am that both my pre-existing .zshrc and .bashrc files are preserved. I humbly request that the Dotly project README.md states that the installation, as it currently stands, deprecates the existing host Zsh and Bash configuration files, but that they are preserved in a backup archive, to mitigate anyone experiencing what I have.

I have already set up a weekly automated Zsh script with Crontab, that automatically copies the current Zsh config, to a dedicated Zsh config archive, in the new Dotly file system. This establishes a weekly snapshot history of the Zsh configuration file, which I should have done long before now. So, this experience has brought about a process that really should have already been in place, which, is a gift. I note that this will be double-handling, when I get in the habit of commiting the Dotly dotfile archive, as recommended by the Dotly framework, to my dedicated GitHub Dotfiles repository. But, I feel, this just brings certainly, that I will never potentially lose aspects of the hard-won-wisdom, facility and utility, of my Zsh shell configuration file. Now, all I have to do, is find the Dotly .zshrc file, in the Dotly repository, and transpose and incorporate its salient logic and code, to my own Zsh and Bash configuration files.

Just to let you know, I am writing a book on POSIX, Bash and Zsh shell scripting, particularly with Termux on Android, for the newly digitally included, out of the vast number of digitally excluded. This will potentially make a lot of people aware of Dotly, and implement the framework. I am going to write of this experience, as a teaching tool of what not to do. That when installing any third-party script, that the user should read the installation script and any executables, in their entirety, and comprehend them, in full, before installing and executing them. 

I have resolved to showcase and endorse Dotly, even though my initial experience was one of terror, frustration and anxiety. Though, I have just installed it, I feel that Dotly is an excellent dotfiles management system, and is going to bring considerable value to my shell user-space, shell scripting, shell workflow, and *NIX environment, particularly in Termux, and wish to bring it to the attention of others.
